jmix-ui/ui/src/main/java/io/jmix/ui/component/filter/builder/PropertyConditionBuilder.java

```diff
@@ -1,156 +1,169 @@
+                                    Messages messages,
+                                    MessageTools messageTools,
+                                    PropertyFilterSupport propertyFilterSupport,
+                                    FilterMetadataTools filterMetadataTools, SingleFilterSupport singleFilterSupport) {
+        this.filterComponents = filterComponents;
+        this.messages = messages;
+        this.messageTools = messageTools;
+        this.propertyFilterSupport = propertyFilterSupport;
+        this.filterMetadataTools = filterMetadataTools;
+        this.singleFilterSupport = singleFilterSupport;
+    }
+
+    @Override
+    public List<FilterCondition> build(Filter filter) {
+        MetaClass filterMetaClass = filter.getDataLoader().getContainer().getEntityMetaClass();
+        String query = filter.getDataLoader().getQuery();
+        Predicate<MetaPropertyPath> propertiesFilterPredicate = filter.getPropertiesFilterPredicate();
+        List<MetaPropertyPath> paths;
+        if(filter.getPropertiesFilterPredicate() instanceof FilterPropertiesInspector && ((FilterPropertiesInspector) filter.getPropertiesFilterPredicate()).getIncludedProperties().size() > 0){
+            paths = filterMetadataTools.getPropertyPathsFromIncludedProperties(filterMetaClass, query,
+                    propertiesFilterPredicate, ((FilterPropertiesInspector) filter.getPropertiesFilterPredicate()).getIncludedProperties());
+        } else {
+            paths = filterMetadataTools.getPropertyPaths(filterMetaClass, query,
+                    propertiesFilterPredicate);
+        }
+
+        return !paths.isEmpty()
+                ? createFilterConditionsByPaths(paths)
+                : Collections.emptyList();
+    }
+
+    @Override
+    public int getOrder() {
+        return JmixOrder.HIGHEST_PRECEDENCE;
+    }
+
+    protected List<FilterCondition> createFilterConditionsByPaths(List<MetaPropertyPath> paths) {
+        List<FilterCondition> conditions = new ArrayList<>();
+
+        HeaderFilterCondition propertiesHeaderCondition = createHeaderFilterCondition(
+                messages.getMessage(PropertyConditionBuilder.class, "propertyConditionBuilder.headerCaption"));
+        conditions.add(propertiesHeaderCondition);
+
+        Map<FilterCondition,MetaPropertyPath> propertyMap = new HashMap<>();
+
+        for (MetaPropertyPath path : paths) {
+            FilterCondition condition = createFilterConditionByPath(path);
+            conditions.add(condition);
+            propertyMap.put(condition,path);
+        }
+
+        for(FilterCondition condition: conditions){
+            MetaPropertyPath path = propertyMap.get(condition);
+            if(path == null) {
+                continue;
+            }
+            FilterCondition parent = path.isDirectProperty()
+                    ? propertiesHeaderCondition
+                    : getParentCondition(path, conditions);
+            condition.setParent(parent);
+        }
```

Comment:
Set the parents after creating the FilterConditions. We do this because the paths returned from `includedProperties` may not be in order of depth as the `include` paths are